cram/cram_codecs.c

```diff
@@ -3095,7 +3095,6 @@ cram_codec *cram_huffman_encode_init(cram_stats *st,
         vals[nvals] = i;
         freqs[nvals] = st->freqs[i];
         assert(st->freqs[i] > 0);
-        ntot += freqs[nvals];
```

Comment:
And it would be good if @jkbonfield can confirm that this wasn't being ignored by mistake.

Reply:
The original code had this after the main loop:

`assert(ntot == st->nsamp);`

It looks like it was basically debugging code during development that was culled incompletely.  We could of course add a check back and handle the error in a more graceful way, but it's been working this long just fine so it seems rather pointless.  

Furthermore, the entire huffman encode-side is now basically defunct, along with most of the other CORE bit-shoveling techniques.  We write to "external" blocks and use rANS instead as this is fast and decouples the streams from one another permitting selective decoding.  Maybe one day if we ever release CRAM 4 we can reenable the bit encoders, directing their output somewhere other than CORE block, but it would break CRAM 3 for now.

The decoder is still needed as we may find crams that use them, but for encoding the only bit of huffman we (ab)use is as a constant value encoder by generating a huffman tree with 1 symbol having a code length of 0 bits.  (Plus we don't actually use the general purpose huffman decoder then as it has a custom constant-value handler.)

Reply:
Thanks.  Good to be sure it wasn't supposed to be there.